File: regalaya-web/src/services/contact.service.ts
import { http } from '@/lib/api'
import type {
  Contact,
  ContactDetail,
  CreateContactRequest,
  UpdateContactRequest,
  CreateSpecialDateRequest,
  UpdateSpecialDateRequest,
  SpecialDate,
  ImportContactEntry,
  ImportReport,
} from '@/types/user'

export const contactService = {
  getAll: (search?: string) => {
    const params = search ? { search } : undefined
    return http.get<Contact[]>('/contacts', params)
  },

  getById: (id: string) =>
    http.get<ContactDetail>(`/contacts/${id}`),

  create: (data: CreateContactRequest) =>
    http.post<Contact>('/contacts', data),

  update: (id: string, data: UpdateContactRequest) =>
    http.put<Contact>(`/contacts/${id}`, data),

  delete: (id: string) =>
    http.delete(`/contacts/${id}`),

  importContacts: (contacts: ImportContactEntry[]) =>
    http.post<ImportReport>('/contacts/import', { contacts }),

  addSpecialDate: (contactId: string, data: CreateSpecialDateRequest) =>
    http.post<SpecialDate>(`/contacts/${contactId}/special-dates`, data),

  updateSpecialDate: (contactId: string, dateId: string, data: UpdateSpecialDateRequest) =>
    http.put<SpecialDate>(`/contacts/${contactId}/special-dates/${dateId}`, data),

  deleteSpecialDate: (contactId: string, dateId: string) =>
    http.delete(`/contacts/${contactId}/special-dates/${dateId}`),
}

File: regalaya-api/src/main/java/br/com/regalaya/contact/services/service/ContactService.java
package br.com.regalaya.contact.services.service;

import br.com.regalaya.contact.dto.requests.CreateContactRequest;
import br.com.regalaya.contact.dto.requests.CreateSpecialDateRequest;
import br.com.regalaya.contact.dto.requests.ImportContactEntry;
import br.com.regalaya.contact.dto.requests.UpdateContactRequest;
import br.com.regalaya.contact.dto.requests.UpdateSpecialDateRequest;
import br.com.regalaya.contact.dto.responses.ContactDetailResponse;
import br.com.regalaya.contact.dto.responses.ContactResponse;
import br.com.regalaya.contact.dto.responses.ImportReportResponse;
import br.com.regalaya.contact.dto.responses.SpecialDateResponse;

import java.util.List;
import java.util.UUID;

public interface ContactService {

    ContactResponse create(UUID userId, CreateContactRequest request);

    ContactResponse update(UUID userId, UUID contactId, UpdateContactRequest request);

    void delete(UUID userId, UUID contactId);

    ContactDetailResponse findById(UUID userId, UUID contactId);

    List<ContactResponse> findAllByUser(UUID userId);

    List<ContactResponse> searchByUser(UUID userId, String query);

    ImportReportResponse importContacts(UUID userId, List<ImportContactEntry> entries);

    SpecialDateResponse addSpecialDate(UUID userId, UUID contactId, CreateSpecialDateRequest request);

    SpecialDateResponse updateSpecialDate(UUID userId, UUID contactId, UUID dateId, UpdateSpecialDateRequest request);

    void deleteSpecialDate(UUID userId, UUID contactId, UUID dateId);
}
